src/app/retrospective/sprint-member-summary/sprint-member-summary.component.ts

```diff
@@ -535,12 +547,47 @@ export class SprintMemberSummaryComponent implements OnInit, OnChanges, OnDestro
         if (this.gridApi) {
             this.gridApi.setFilterModel(null);
             this.gridApi.onFilterChanged();
-            this.filterService.setFilterData(RETRO_SUMMARY_TYPES.MEMBER, this.gridApi.getFilterModel());
+            this.saveFilterState();
         }
     }
+    // TO save the states of column filters
+    saveFilterState() {
+        this.gridService.saveFilterState(RETRO_SUMMARY_TYPES.MEMBER, this.gridApi.getFilterModel());
 
+    }
     // restore the state of column filters
     restoreFilterData() {
-        this.gridApi.setFilterModel(this.filterService.getFilterData(RETRO_SUMMARY_TYPES.MEMBER));
+        this.gridApi.setFilterModel(this.gridService.getFilterState(RETRO_SUMMARY_TYPES.MEMBER));
+    }
+    // To save the current state of columns in angular scope
+    saveColumnState(columnState) {
+        if (this.setColumnflag && this.isTabActive) {
+            const savedColumnState = this.gridService.getColumnState(this.retrospectiveID, RETRO_SUMMARY_TYPES.MEMBER);
+            if (!this.isSprintEditable && savedColumnState && columnState.length < savedColumnState.length) {
+                columnState = this.addDeleteColumnState(columnState, savedColumnState);
+            }
+            this.gridService.saveColumnState(this.retrospectiveID, RETRO_SUMMARY_TYPES.MEMBER, columnState);
+        }
+        this.setColumnflag = true;
+    }
+    // To restore the saved state of columns
+    applyColumnState() {
+        let columnState = this.gridService.getColumnState(this.retrospectiveID, RETRO_SUMMARY_TYPES.MEMBER);
+        if (columnState && columnState.length > 0) {
+            const currentColumnState = this.columnApi.getColumnState();
+            if (this.isSprintEditable && columnState.length < currentColumnState.length) {
+                columnState = this.addDeleteColumnState(columnState, currentColumnState);
```

Comment:
Parameters are being passed wrong. Shouldn't it be like this:
```suggestion
                columnState = this.addDeleteColumnState(currentColumnState, columnState);
```

Reply:
done